helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -185,17 +185,13 @@ public void updatePreviousAssignedTasksStatus(
         switch (currState) {
         case RUNNING: {
           TaskPartitionState nextState = TaskPartitionState.RUNNING;
-          if (jobState == TaskState.TIMING_OUT) {
+          if (jobState == TaskState.TIMING_OUT || jobState == TaskState.TIMED_OUT
+              || jobState == TaskState.FAILING || jobState == TaskState.FAILED
+              || jobState == TaskState.ABORTED) {
```

Comment:
Would it be cleaner to define a set of enums (states) where we should mark the tasks aborted? Then we could do a simple exists() check. 

Or, it seems that if we check for != condition, we won't have to list out so many states here?

Reply:
Also just to confirm, this is to make sure that task states that are stuck in RUNNING get updated to ABORTED?

Reply:
Done. I create a list for these states. Yes and also make sure task goes to ABORTED state and we release quota.